Visualize process execution paths

Nijin22 opened the pull request.

Nijin22 committed:
Extract flows from database

Nijin22 committed:
Increase precision of timestamps to avoid problems with events being tracked at the same second

Nijin22 committed:
Add counting number of following events, for each event

Nijin22 committed:
Add generating json files for d3-sankey-diagrams

Nijin22 committed:
Add partial rendering of sankey diagram

Nijin22 committed:
Improve readability of graph

Nijin22 committed:
Add respecting "timeFrom" and "timeUntil" parameters

Nijin22 committed:
Remove outdated TODO. The sankey plugin is not compatible with this version of d3

Nijin22 committed:
Add simple favicon

Mostly just so we don't get any of those browser-induced 404 errors in the log.

Nijin22 committed:
Fix "undefined" appearing on hover over links

Nijin22 committed:
Add pseudo-random colors for links

Nijin22 committed:
Add benchmark statements for process flow analysis

Nijin22 committed:
Add Unit test for flow endpoint

Nijin22 committed:
Remove "Authentication & Authorization" from API documentation

as it explained manual steps that can be done via the correct
endpoints instead.

Nijin22 committed:
Add API documentation of the two new endpoints

Nijin22 committed:
Remove outdated TODOs

sweipert96 commented on src/ratpack/resources/openapi.yaml:
```diff
@@ -7,13 +7,6 @@ info:
     A service useful for tracking various events in Serviceportal processes and to generate various
     reports about these events.
     <br><br>
```
Da der Abschnitt "Authentication & Authorization" weg ist, können die line breaks hier auch weg, oder? 🙂

Nijin22 committed:
Remove unnecessary line breaks

Resolves https://github.com/Seitenbau/service-kommune-tracker/pull/16#discussion_r896709463
